packages/react-native/UPGRADING.md

```diff
@@ -0,0 +1,279 @@
+
+### Adding `@bugsnag/react-native`
+
+Full instructions are available at the [React Native integration guide](https://docs.bugsnag.com/platforms/react-native/react-native#installation) but here is a summary.
+
+Install the new package from npm using your preferred command:
+
+```sh
+npm install --save @bugsnag/react-native
+# or
+yarn add @bugsnag/react-native
+```
+
+#### Android
+
+Add the following to your `android/app/build.gradle` to integrate Bugsnag into your Anroid project:
+
+```groovy
+apply from: "../../node_modules/@bugsnag/react-native/bugsnag-react-native.gradle"
+```
+
+#### iOS
+
+Run cocoapods again so that it picks up the new local dependency:
+
+```
+cd ios
+pod install
+```
+
+### Updating usage
+
+Full integration instructions are available on the [React Native integration guide](https://docs.bugsnag.com/platforms/react-native/react-native#installation). This section will focus on specific areas that have changed.
+
+Since this update involved a major bump to Bugsnag's JS, Android and iOS notifiers – bringing all of their APIs in sync, virtually every usage of Bugsnag's API in your application will need updating.
+
+#### Initialization
+
+Previously in `bugsnag-react-native` it was possible to initialize in a variety of ways. Now there is one way to do it.
+
+In each of your native projects you need to initialize Bugsnag. This was optional before, but it means that now crashes before the JS layer loads will always be caught.
+
+To facilitate this, and to get Bugsnag started as early as possible, most configuration can only be done in the native layer. Some JS-specific options are allowed in the JS layer.
+
+##### iOS
+
+Update any of the following imports in. These will likely be in `AppDelegate.m` where it's recommended to configure Bugsnag, but may be in other locations too:
+
+```diff
+- #import <BugsnagReactNative/BugsnagReactNative.h>
+- #import "BugsnagConfiguration.h"
++ import <Bugsnag/Bugsnag.h>
+```
+
+Inside your `application:didFinishLaunchingWithOptions` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- [BugsnagReactNative start];
++ [Bugsnag start];
+```
+
+Or if you need to customize your configuration:
+
+```diff
+- BugsnagConfiguration *config = [BugsnagConfiguration new];
+- config.releaseStage = @"beta";
+- [BugsnagReactNative startWithConfiguration:config];
++ BugsnagConfiguration *config = [BugsnagConfiguration loadConfig];
++ config.releaseStage = @"beta";
++ [Bugsnag startWithConfiguration:config];
+```
+
+See the new [Bugsnag Cocoa configuration guide](https://docs.bugsnag.com/platforms/ios/configuration-options/) for all available configuration options.
+
+If you have set your API key in `Info.plist`, the format has changed from a top level string to a dictionary.
+
+```
+- <key>BugsnagAPIKey</key>
+- <string>YOUR_API_KEY</string>
++ <key>bugsnag</key>
++ <dict>
++     <key>apiKey</key>
++     <string>YOUR_API_KEY</string>
++ </dict>
+```
```

Comment:
We want to emphasise use of the plist/manifest as much as possible - can you move this to after the `[Bugsnag start]` snippet and reference the new ability to set other configuration options? Then have the in-code snippet afterwards.